core/embed/boardloader/main.c

```diff
@@ -229,18 +241,20 @@ int main(void) {
   }
 #endif
 
-  image_header hdr;
+  const image_header *hdr =
+      read_image_header((const uint8_t *)BOOTLOADER_START,
+                        BOOTLOADER_IMAGE_MAGIC, BOOTLOADER_IMAGE_MAXSIZE);
+
+  ensure(hdr != NULL ? sectrue : secfalse, "invalid bootloader header");
```

Comment:
again, by corrupting `hdr` a glitcher can continue to `check_image_header_sig` even if `read_image_header` returns an error.

would this even be a problem? a corrupted `hdr` is pointing to random memory, in the typical case outside the address space, right? so the chance of `check_image_header_sig` doing anything besides crashing is nil, right?

otoh checking that hdr == BOOTLOADER_START is a cheap way to avoid the issue in the first place without any obvious downsides

Reply:
fixed https://github.com/trezor/trezor-firmware/pull/2649/commits/0c06166e66b15dc84c3f5ce1166ab7e7b36a028b